### PR TITLE
Enum serialisation for NetPacket

### PR DESCRIPTION
**Pull request by carlst99**
NOTE: Unfortunately, this can't be used in a .NET Standard 1.3 environment, as [Type.IsEnum](https://docs.microsoft.com/en-gb/dotnet/api/system.type.isenum?view=netstandard-2.0#System_Type_IsEnum) is not available.

However, it's not all bad 😄. I've added Enum read and write capabilities to NetPacketStream, allowing an enum to be passed straight to the Read() and Write() methods, without worrying about casting. I've also added test methods.

### CODE REVIEW

**Comment by FalconEye36** [comment]
After getting the unit tests running I'm having a few issues. I'm going to close this for now

**Comment by FalconEye36** [comment]
Looks like tests were failing because the normal test project targets NET Core 1.0. I've shifted the enum tests into a new test project targeting NET Core 2.0 and they passed without incident.

File: src/Ether.Network/Packets/NetPacketStream.cs [hunk 1]
```diff
@@ -58,6 +58,13 @@ public virtual T Read<T>()
 
             if (NetPacketMethods.ReadMethods.ContainsKey(type))
                 return (T)NetPacketMethods.ReadMethods[type](this._memoryReader);
+#if NETSTANDARD2_0
+            else if (type.IsEnum && NetPacketMethods.WriteMethods.ContainsKey(type.GetEnumUnderlyingType()))
```
Comment (Eastrall) [inline src/Ether.Network/Packets/NetPacketStream.cs]: You are checking on the `WriteMethods` dictionnary instead of `ReadMethods`.

Comment (FalconEye36) [reply src/Ether.Network/Packets/NetPacketStream.cs]: Woops. Thanks 👍

**Comment by FalconEye36** [comment]
I've just double check, Type.IsEnum isn't available in NetStandard 1.4 [(see here)](https://docs.microsoft.com/en-gb/dotnet/api/system.type.isenum?view=netframework-4.7.1#applies-to). It was first introduced in 2.0. Were you referring to something else?

**Comment by Eastrall** [comment]
I was refering to the [Net Standard 1.4](https://github.com/dotnet/standard/blob/master/docs/versions/netstandard1.4_ref.md) API reference.
But I think this feature should be implemented in a major version (V4.0). Because, like you said, if developers will have some issues with compilation.
I let the Pull Request open until then 😄

**Comment by shmutalov** [comment]
Any news?

**Comment by Eastrall** [comment]
@shmutalov this pull request needs to update the .NET Standard version from 1.3 to 1.4. We cannot do this right in versions 3.X because, it will break project compatibility with 3.X series. We need to wait for version 4.X of Ether in order to introduce this new feature.

**Pull request closed without merging**